libs/SmartSync/SmartSync/Classes/SFBatchingSyncUpTarget.h

```diff
@@ -34,12 +34,15 @@ NS_SWIFT_NAME(BatchingSyncUpTarget)
  */
 @interface SFBatchingSyncUpTarget : SFSyncUpTarget <SFAdvancedSyncUpTarget>
 
-/** Factory methods
+/** Constructor
  */
+- (instancetype)initWithCreateFieldlist:(nullable NSArray *)createFieldlist
```

Comment:
@wmathurin Could we  specify the type in the list  like below?
``` (instancetype)initWithCreateFieldlist:(nullable NSArray<NSString *> *)createFieldlist  updateFieldlist:(nullable NSArray<NSString *> *)updateFieldlist maxBatchSize:(nullable NSNumber *)maxBatchSize; ```

Reply:
Does this need `NS_SWIFT_NAME` magic too?

Reply:
It's on line 31.

Reply:
Here is the generated swift header
```swift
open class BatchingSyncUpTarget : SyncUpTarget, AdvancedSyncUpTarget {

    
    /** Constructor
     */
    public init(createFieldlist: [Any]?, updateFieldlist: [Any]?, maxBatchSize: NSNumber?)

    
    
    /** Factory method
     */
    open class func build(dict: [AnyHashable : Any]?) -> Self
}
```